index.js

```diff
@@ -28,10 +37,184 @@ for (const file of commandFiles) {
+		.setDescription(`**Subject:** ${subject} \n**Level:** ${level} \n**Class(es) per week:** ${frequency} \n**Pay per class:** ${money} \n**Time slots:** ${availabilities.join(", ")}\n**Class duration**: ${classDuration} hour(s)`)
+		.setTimestamp() 
+		.setFooter({ text: 'Please select the date and time that fits you best and we will get back to you on the next steps.', iconURL:'https://i.imgur.com/i1k870R.png'});
+
+	//Create a row object that will hold the select menu
+	const row = new ActionRowBuilder();
+
+	//Create Select Menu object
+	const menu = new SelectMenuBuilder()
+				.setCustomId("dateSelection")
+				.setPlaceholder(`Please select ${frequency} date option(s)`)
+				.setMinValues(frequency)
+				.setMaxValues(frequency)
+
+	//Loop through the availabilities' list and display every ability to tutor via select menu 
+	availabilities.forEach((dateAndTime) => {
+		menu.addOptions(
+			{
+				"label": dateAndTime,
+				"value": dateAndTime
+			}
+		)
+	});
+
+	//Add menu to row
+	row.addComponents(menu);
+
+	// Add submit button inside new row
+	const row2 = new ActionRowBuilder()
+		.addComponents( 
+			new ButtonBuilder()
+				.setCustomId("submitButton")
+				.setLabel('Submit request')
+				.setStyle(ButtonStyle.Success),
+
+			new ButtonBuilder()
+				.setCustomId("cancelButton")
+				.setLabel('Cancel request')
+				.setStyle(ButtonStyle.Danger),
+
+		);
+			
+	// Sends both objects to channel
+	channel.send({ embeds: [msgEmbed], components: [row,row2]});
+}
+
+
+client.on('interactionCreate', async interaction => {
+	
+	// The answers of all tutors that are stored inside the CSV in the form of an array
+	const answers = await csv().fromFile("answers.csv");
+
+	if (interaction.customId === 'dateSelection') {
+	
+		// Checks if the tutor has already an answer stored inside then CVS
+		if (tutorMadeASelection(interaction.user.id, answers)) {
+			interaction.reply({ content: "Please **submit** or **cancel** your first request to proceed.", ephemeral: true });
+		} 
+		else {
+			// Pushes the latest answer 
+			answers.push({ tutorId: interaction.user.id, selection: interaction.values.toString() });
+
+			//Writes the modifications in the CSV file 
+			fsCsv.writeFileSync("answers.csv", new Parser({fields: ["tutorId", "selection"] }).parse(answers));
+
+			interaction.reply({ content: "If you're done with your selection, please submit. You can still change your selection.", ephemeral: true });
+		}
+		
+	}
+
+	if (interaction.customId === 'submitButton') {
+
+		if (tutorMadeASelection(interaction.user.id, answers)) {
+
+			// POST request to API is created with tutorId and selection under tutorDemand route
+
+			interaction.reply({content: "Your request has been sent.", ephemeral: true});
+
+			// Delete the appropriate line in the CSV and write the new CSV state
+			fsCsv.writeFileSync("answers.csv", new Parser({fields: [ "tutorId", "selection"] }).parse(deleteTutorAnswer(interaction.user.id, answers)));
```

Comment:
where are the answers read from the csv?

Reply:
The deleteTutor function returns an array storing the updated state of the answers in the form of a list. This list is then written to the CSV.